lib/stdlib/src/timer.erl

```diff
@@ -203,250 +269,216 @@ tc(M, F, A) ->
-%% {Time, BRef, Op, MFA}, where Time is in microseconds.
-%% Returns {Timeout, Timers}, where Timeout is in milliseconds.
-%%
-timer_timeout(SysTime) ->
-    case ets:first(?TIMER_TAB) of
-	'$end_of_table' -> 
-	    infinity;
-	{Time, _Ref} when Time > SysTime ->
-	    Timeout = (Time - SysTime + 999) div 1000,
-	    %% Returned timeout must fit in a small int
-	    erlang:min(Timeout, ?MAX_TIMEOUT);
-	Key ->
-	    case ets:lookup(?TIMER_TAB, Key) of
-		[{Key, timeout, MFA}] ->
-		    ets:delete(?TIMER_TAB,Key),
-		    do_apply(MFA),
-		    timer_timeout(SysTime);
-		[{{Time, Ref}, Repeat = {repeat, Interv, To}, MFA}] ->
-		    ets:delete(?TIMER_TAB,Key),
-		    NewTime = Time + Interv,
-		    %% Update the interval entry (last in table)
-		    ets:insert(?INTERVAL_TAB,{{interval,Ref},{NewTime,Ref},To}),
-		    do_apply(MFA),
-		    ets:insert(?TIMER_TAB, {{NewTime, Ref}, Repeat, MFA}),
-		    timer_timeout(SysTime)
-	    end
-    end.
-
-%%
-%% delete_ref 
-%%
-
-delete_ref(BRef = {interval, _}) ->
-    case ets:lookup(?INTERVAL_TAB, BRef) of
-	[{_, BRef2, _Pid}] ->
-	    ets:delete(?INTERVAL_TAB, BRef),
-	    ets:delete(?TIMER_TAB, BRef2);
-	_ -> % TimerReference does not exist, do nothing
+    {ok, Tab}.
+
+%% Remove a timer.
+remove_timer(TRef, Tab) ->
+    _ = case ets:take(Tab, TRef) of
+	[{TRef, SRef}] ->
+	    erlang:cancel_timer(SRef, [{async, true}, {info, false}]);
+	[] -> % TimerReference does not exist, do nothing
 	    ok
-    end;
-delete_ref(BRef) ->
-    ets:delete(?TIMER_TAB, BRef).
-
-%%
-%% pid_delete
-%%
-
--spec pid_delete(pid()) -> 'ok'.
-pid_delete(Pid) ->
-    IntervalTimerList = 
-	ets:select(?INTERVAL_TAB,
-		   [{{'_', '_','$1'},
-		     [{'==','$1',Pid}],
-		     ['$_']}]),
-    lists:foreach(fun({IntKey, TimerKey, _ }) ->
-			  ets:delete(?INTERVAL_TAB, IntKey),
-			  ets:delete(?TIMER_TAB, TimerKey)
-		  end, IntervalTimerList).
-
-%% Calculate time to the next timeout. Returned timeout must fit in a 
-%% small int.
-
--spec next_timeout() -> timeout().
-next_timeout() ->
-    case ets:first(?TIMER_TAB) of
-	'$end_of_table' -> 
-	    infinity;
-	{Time, _} ->
-	    erlang:min(positive((Time - system_time() + 999) div 1000), ?MAX_TIMEOUT)
-    end.
+    end,
+    ok.
 
 %% Help functions
-do_apply({M,F,A}) ->
-    case {M, F, A} of
-	{?MODULE, send, A} -> 
-	    %% If send op. send directly, (faster than spawn)
-	    catch send(A);
-	{erlang, exit, [Name, Reason]} ->
-	    catch exit(get_pid(Name), Reason);
-	_ -> 
-	    %% else spawn process with the operation
-	    catch spawn(M,F,A)      
-    end.
-
-positive(X) ->
-    erlang:max(X, 0).
 
+%% If send op. send directly, (faster than spawn)
+do_apply({?MODULE, send, A}) ->
+    catch send(A);
+%% If exit op. resolve registered name
+do_apply({erlang, exit, [Name, Reason]}) ->
+    catch exit(get_pid(Name), Reason);
+do_apply({M,F,A}) ->
+    catch spawn(M,F,A).
 
-%%
-%%  system_time() -> time in microseconds
```

Comment:
I suggest that you add a comment that explains what this function does. It is not immediately obvious from the code what is returned.

Reply:
Added in last commit.